cpp/doxygen/developer_guide/DOCUMENTATION.md

```diff
@@ -0,0 +1,537 @@
+# libcuspatial C++ Documentation Guide
```

Comment:
Curious why this file is not labeled as `rename` but as addition. Assuming content is unchanged besides @thomcom 's [comment](https://github.com/rapidsai/cuspatial/pull/673/files#r975802135)